pytket/binders/circuit/Circuit/main.cpp

```diff
@@ -414,7 +414,15 @@ void def_circuit(py::class_<Circuit, std::shared_ptr<Circuit>> &pyCircuit) {
           "n_gates", &Circuit::n_gates,
           ":return: the number of gates in the Circuit")
       .def_property_readonly(
-          "wasm_uid", &Circuit::get_wasm_file_uid,
+          "wasm_uid",
+          [](const Circuit &circ) {
+            std::optional<std::string> result;
+            try {
+              result = circ.get_wasm_file_uid();
+            } catch (const std::exception &) {
```

Comment:
Can we propagate the change to the C++, i.e. male `get_wasm_file_uid()` return a `std::optional<std::string>`? I think that's better than a catch-all exception handler, and could actually make the C++ code nicer too.

Reply:
done in [00c3b30](https://github.com/CQCL/tket/pull/1691/commits/00c3b307346d3d7b91dc36d0fef2158835227bbb)